keanu-project/src/main/java/io/improbable/keanu/tensor/TensorShape.java

```diff
@@ -199,6 +199,28 @@ public static boolean isLengthOne(long[] shape) {
         return (shape1.length >= shape2.length) ? shape1 : shape2;
     }
 
+    public static long[] getBroadcastResultShape(long[] left, long[] right) {
+
+        final long[] highestRank = left.length > right.length ? left : right;
+        long[] resultShape = Arrays.copyOf(highestRank, highestRank.length);
+
+        int lowestRank = Math.min(left.length, right.length);
+        for (int i = 1; i <= lowestRank; i++) {
```

Comment:
why not call `public static boolean isBroadcastable(long[] left, long[] right)`?

Reply:
Only because that would require iterating through the shape twice. Sometimes you want to know if it's broadcastable without throwing an exception and sometimes you want to get the broadcasted shape or throw an exception. You could check isBroadcastable before getting the broadcastable shape but that would iterate twice, which probably wouldn't be a big deal.

Reply:
It would be nice to get rid of the code duplication though. How about if `isBroadcastable` were to call `getBroadcastShape`?

Reply:
👍